src/names.c

```diff
@@ -0,0 +1,114 @@
+    [NETPLAN_BACKEND_NETWORKD] = "networkd",
+    [NETPLAN_BACKEND_NM] = "NetworkManager",
+    [NETPLAN_BACKEND_OVS] = "OpenVSwitch",
+};
+
+static const char* const
+netplan_wifi_mode_to_str[NETPLAN_WIFI_MODE_MAX_] = {
+    [NETPLAN_WIFI_MODE_INFRASTRUCTURE] = "infrastructure",
+    [NETPLAN_WIFI_MODE_ADHOC] = "adhoc",
+    [NETPLAN_WIFI_MODE_AP] = "ap",
+    [NETPLAN_WIFI_MODE_OTHER] = NULL,
+};
+
+
+static const char* const
+netplan_def_type_to_str[NETPLAN_DEF_TYPE_MAX_] = {
+    [NETPLAN_DEF_TYPE_NONE] = NULL,
+    [NETPLAN_DEF_TYPE_ETHERNET] = "ethernets",
+    [NETPLAN_DEF_TYPE_WIFI] = "wifis",
+    [NETPLAN_DEF_TYPE_MODEM] = "modems",
+    [NETPLAN_DEF_TYPE_BRIDGE] = "bridges",
+    [NETPLAN_DEF_TYPE_BOND] = "bonds",
+    [NETPLAN_DEF_TYPE_VLAN] = "vlans",
+    [NETPLAN_DEF_TYPE_TUNNEL] = "tunnels",
+    [NETPLAN_DEF_TYPE_PORT] = NULL,
+    [NETPLAN_DEF_TYPE_NM] = "nm-devices",
+};
+
+static const char* const
+netplan_auth_key_management_type_to_str[NETPLAN_AUTH_KEY_MANAGEMENT_MAX] = {
+    [NETPLAN_AUTH_KEY_MANAGEMENT_NONE] = "none",
+    [NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK] = "psk",
+    [NETPLAN_AUTH_KEY_MANAGEMENT_WPA_EAP] = "eap",
+    [NETPLAN_AUTH_KEY_MANAGEMENT_8021X] = "802.1x",
+};
+
+static const char* const
+netplan_auth_eap_method_to_str[NETPLAN_AUTH_EAP_METHOD_MAX] = {
+    [NETPLAN_AUTH_EAP_NONE] = NULL,
+    [NETPLAN_AUTH_EAP_TLS] = "tls",
+    [NETPLAN_AUTH_EAP_PEAP] = "peap",
+    [NETPLAN_AUTH_EAP_TTLS] = "ttls",
+};
+
+static const char* const
+netplan_tunnel_mode_to_str[NETPLAN_TUNNEL_MODE_MAX_] = {
+    [NETPLAN_TUNNEL_MODE_UNKNOWN] = NULL,
+    [NETPLAN_TUNNEL_MODE_IPIP] = "ipip",
+    [NETPLAN_TUNNEL_MODE_GRE] = "gre",
+    [NETPLAN_TUNNEL_MODE_SIT] = "sit",
+    [NETPLAN_TUNNEL_MODE_ISATAP] = "isatap",
+    [NETPLAN_TUNNEL_MODE_VTI] = "vti",
+    [NETPLAN_TUNNEL_MODE_IP6IP6] = "ip6ip6",
+    [NETPLAN_TUNNEL_MODE_IPIP6] = "ipip6",
+    [NETPLAN_TUNNEL_MODE_IP6GRE] = "ip6gre",
+    [NETPLAN_TUNNEL_MODE_VTI6] = "vti6",
+    [NETPLAN_TUNNEL_MODE_GRETAP] = "gretap",
+    [NETPLAN_TUNNEL_MODE_IP6GRETAP] = "ip6gretap",
+    [NETPLAN_TUNNEL_MODE_WIREGUARD] = "wireguard",
+};
+
+static const char* const
+netplan_addr_gen_mode_to_str[NETPLAN_ADDRGEN_MAX] = {
+    [NETPLAN_ADDRGEN_DEFAULT] = NULL,
+    [NETPLAN_ADDRGEN_EUI64] = "eui64",
+    [NETPLAN_ADDRGEN_STABLEPRIVACY] = "stable-privacy"
+};
+
+#define NAME_FUNCTION(_radical, _type) const char *netplan_ ## _radical ## _name( _type val) \
+{ \
+    return (val < sizeof( netplan_ ## _radical ## _to_str )) ?  netplan_ ## _radical ## _to_str [val] : NULL; \
+}
+
+NAME_FUNCTION(backend, NetplanBackend);
+NAME_FUNCTION(def_type, NetplanDefType);
+NAME_FUNCTION(auth_key_management_type, NetplanAuthKeyManagementType);
+NAME_FUNCTION(auth_eap_method, NetplanAuthEAPMethod);
+NAME_FUNCTION(tunnel_mode, NetplanTunnelMode);
+NAME_FUNCTION(addr_gen_mode, NetplanAddrGenMode);
+NAME_FUNCTION(wifi_mode, NetplanWifiMode);
+
+/* ABI compatibility definitions */
+
+const char *
```

Comment:
```suggestion
const char*
```
Just a tiny nitpick...